test/unit/rounding/rounding.cpp

```diff
@@ -123,4 +123,80 @@ namespace {
     }
+
+    namespace tie_to_pos_inf_rounding {
+
+        namespace convert {
+            static_assert(
+                    identical(
+                            cnl::uint8{101},
+                            cnl::convert<cnl::tie_to_pos_inf_rounding_tag, cnl::_impl::native_tag, cnl::uint8>(100.5)),
+                    "cnl::convert test failed");
+            static_assert(
+                    identical(
+                            cnl::int16{-1000},
+                            cnl::convert<cnl::tie_to_pos_inf_rounding_tag, cnl::_impl::native_tag, cnl::int16>(
+                                    -1000.5L)),
+                    "cnl::convert test failed");
+            static_assert(
+                    identical(
+                            55,
+                            cnl::convert<cnl::tie_to_pos_inf_rounding_tag, cnl::_impl::native_tag, cnl::int32>(55.2F)),
+                    "cnl::convert test failed");
+            static_assert(
+                    identical(
+                            -0,
+                            cnl::convert<cnl::tie_to_pos_inf_rounding_tag, cnl::_impl::native_tag, int>(-0.50)),
+                    "cnl::convert test failed");
+            static_assert(
+                    identical(
+                            -0,
+                            cnl::convert<cnl::tie_to_pos_inf_rounding_tag, cnl::_impl::native_tag, int>(-0.49)),
+                    "cnl::convert test failed");
+            static_assert(
+                    identical(
+                            +0,
+                            cnl::convert<cnl::tie_to_pos_inf_rounding_tag, cnl::_impl::native_tag, int>(0.49)),
+                    "cnl::convert test failed");
+            static_assert(
+                    identical(
+                            +1,
+                            cnl::convert<cnl::tie_to_pos_inf_rounding_tag, cnl::_impl::native_tag, int>(0.50)),
+                    "cnl::convert test failed");
+        }
+
+        namespace divide {
+            static_assert(identical(-1, cnl::divide<cnl::tie_to_pos_inf_rounding_tag>(-990, 661)),
+                    "cnl::divide test failed");
+            static_assert(identical(2, cnl::divide<cnl::tie_to_pos_inf_rounding_tag>(-606, -404)),
+                    "cnl::divide test failed");
+            static_assert(identical(1, cnl::divide<cnl::tie_to_pos_inf_rounding_tag>(8, 9)),
+                    "cnl::divide test failed");
+            static_assert(identical(-1, cnl::divide<cnl::tie_to_pos_inf_rounding_tag>(9, -8)),
+                    "cnl::divide test failed");
+            static_assert(identical(-1, cnl::divide<cnl::tie_to_pos_inf_rounding_tag>(-9, 6)),
+                    "cnl::divide test failed");
+            static_assert(identical(1, cnl::divide<cnl::tie_to_pos_inf_rounding_tag>(-9, -7)),
+                    "cnl::divide test failed");
+            static_assert(identical(2, cnl::divide<cnl::tie_to_pos_inf_rounding_tag, cnl::uint16, int>(999, 666)),
```

Comment:
Algorithm actually has change of sign depending on the parameters. Would change of type from cnl::uint16 to cnl::int16 solve related compilation error?